### PR TITLE
Add Morello support to Makefile

### PR DESCRIPTION
**Pull request by drpdr**
Add Morello support to Makefile

- default architecture is Morello
- ENV files to switch to riscv64

### CODE REVIEW

**Comment by drpdr** [comment]
PR Ready for review.

File: Makefile [hunk 1]
```diff
@@ -1,9 +1,8 @@
-CC=$(HOME)/cheri/output/sdk/bin/clang
-CFORMAT=$(HOME)/cheri/output/sdk/bin/clang-format
-CXX=$(HOME)/cheri/output/sdk/bin/clang++
-CFLAGS=-fuse-ld=lld --config cheribsd-riscv64-purecap.cfg
-SSHPORT=10021
-export 
+ifndef ARCH
```
Comment (ltratt) [inline Makefile:1]: I'm pretty sure I'll forget to define `ARCH` and then on risc-v I'll type `make` and be confused at the resulting weird errors! We could either perhaps:

1. use something like `uname` to automatically guess the right architecture
2. have two separate files `Makefile.riscv` and `Makefile.morello` to separate things out in an unambiguous way

I'm fairly neutral between the two.

Comment (drpdr) [reply Makefile:1]: I should SSH into to ``uname`` while cross-compiling. I'll go for the second, then.

Comment (ltratt) [reply Makefile:1]: Should we be cross-compiling on a remote machine? Or should we be saying "actually you need to compile these on a CHERI machine directly"? I would favour the latter (cross compiling always ends in tears!).

Comment (alex4o) [reply Makefile:1]: It is very important to be able to corss-compile imo. As compiling on qemu will take ages.

Comment (ltratt) [reply Makefile:1]: Can we roughly time how long it takes?

Perhaps I've actually over-stated things in one way. I dislike cross-compilation, but I have no problem with making it *possible*. However at the moment I think we *mandate* cross-compilation with the `ssh` stuff?

Comment (alex4o) [reply Makefile:1]: 1. There is no clang compiler on CheriBSD by default. So you need to compile and install it additionally.
2. We do not mandate it. The `ssh` stuff is to run when cross compiling.

Comment (drpdr) [reply Makefile:1]: I didn't mean linking SSH and cross-compiling in my comment above :)

Comment (ltratt) [reply Makefile:1]: OK, I'm making this PR go off-piste, which is bad of me. I think there's a deeper issue here that we can come back to in a different PR.

**Comment by drpdr** [comment]
PR Ready for review.

File: Makefile.riscv64 [hunk 2]
```diff
@@ -2,8 +2,8 @@ CC=$(HOME)/cheri/output/sdk/bin/clang
 CFORMAT=$(HOME)/cheri/output/sdk/bin/clang-format
 CXX=$(HOME)/cheri/output/sdk/bin/clang++
 CFLAGS=-fuse-ld=lld --config cheribsd-riscv64-purecap.cfg
-SSHPORT=10021
-export 
+SSHPORT=10013
```
Comment (ltratt) [inline Makefile.riscv64:5]: Is there any reason to change the port number?

Comment (drpdr) [reply Makefile.riscv64:5]: You are absolutely right. That's my port. I will fix it.

Comment (drpdr) [reply Makefile.riscv64]: Fixed in 56f2d0a.

**Comment by ltratt** [comment]
Please squash.

**Comment by drpdr** [comment]
splat

**Comment by drpdr** [comment]
Right, that's the Makefile in the bot. I'll fix it.

**Comment by drpdr** [comment]
bors try

**Comment by drpdr** [comment]
Fixed in 073b528.

**Comment by ltratt** [comment]
Please squash.

**Comment by drpdr** [comment]
Done.

**Pull request merged**